### PR TITLE
#1168: Incorrect TLS cert instructions for Intermediate CA

### LINKED ISSUE #1168: Incorrect TLS cert configuration instructions for Intermediate CA on UCP
### Problem description

If a customer has an intermediate CA, it should be provided in the CA certificate field not the Certificate field. Otherwise the customer will receive an error updating certs that states `unable to load x509 key pair, the private key does not match the public key`

### Problem location

- I saw a problem on the following URL: https://docs.docker.com/datacenter/ucp/2.0/guides/configuration/#/customize-the-ucp-tls-certificates

### Project version(s) affected

UCP 2.0 and UCP 2.1 

### Suggestions for a fix

`CA certificate` - root CA followed by the intermediate CA
`Certificate` - server public certificate followed by any other intermediates (typically there aren't multiple)
`Key` - private key (make sure it is not encrypted with a password)
for encrypted keys, it should say `ENCRYPTED` in the first line

### PR DESCRIPTION
**Pull request by clocklear**
### Proposed changes

Fixed incorrect TLS cert configuration instructions for Intermediate CA on UCP versions 2.0 and 2.1

### Related issues (optional)

Fixes #1168

### CODE REVIEW

**Comment by joaofnfernandes** [comment]
@clocklear thanks for contributing. Did you get a chance to test this on a real deployment, or you're just following what #1168 tells you to fix?

I'm asking because I have a deployment where
* CA cert- just the CA public cert
* server cert- the public cert for the domain + the public cert for the intermediary (in this order)

Making the CA cert be the CA public cert + intermediary (in this order) also works, but seems redundant.

Were you able to run into a scenario where you had to add the intermediary to the CA cert to make this work?

**Comment by clocklear** [comment]
No, I'm currently unable to test this, as I don't have any apps inside
docker cloud. I followed what #1268 suggested.

Based on what you are suggesting, it seems as though both solutions would
work, but are a bit redundant.

**Comment by joaofnfernandes** [comment]
I'd like to have @programmerq opinion about this. Interestingly enough he will be giving a talk today about troubleshooting techniques, and I think he is going to talk how to set up your certificates in DDC.

File: datacenter/ucp/2.0/guides/configuration/index.md [hunk 1]
```diff
@@ -39,11 +39,13 @@ and click **Certificates**.
 
 Upload your certificates and keys:
 
-* A ca.pem file with the root CA public certificate.
-* A cert.pem file with the TLS certificate and any intermediate CA public
+* A `ca.pem` file with the root CA public certificate followed by any intermediate CA public
```
Comment (joaofnfernandes) [inline datacenter/ucp/2.0/guides/configuration/index.md]: After several tests and a conversation with @programmerq I can confirm that this is what you should do:
* CA - the CA public certificate
* Server cert - The public certificate for your domain, and any intermediate public certificates, in this order.

Can you revert your change to line `42` and update `44` make make this clear?

**Comment by clocklear** [comment]
Absolutely. Not at my computer at the moment, but will address ASAP.

**Comment by joaofnfernandes** [comment]
No need to rush! We're all recovering for DockerCon 😂

File: datacenter/ucp/2.1/guides/admin/configure/use-your-own-tls-certificates.md [hunk 2]
```diff
@@ -39,11 +39,13 @@ and click **Certificates**.
 
 Upload your certificates and keys:
 
-* A ca.pem file with the root CA public certificate.
-* A cert.pem file with the TLS certificate and any intermediate CA public
+* A `ca.pem` file with the root CA public certificate followed by any intermediate CA public
```
Comment (joaofnfernandes) [inline datacenter/ucp/2.1/guides/admin/configure/use-your-own-tls-certificates.md]: Looks like you forgot to update this article for version 2.1. Once that's fixed, we're ready to ship.

**Comment by clocklear** [comment]
Whoops.  Sorry 'bout that.  Forgot it was in two places.

**Pull request merged**